pkg/applyset/applyset.go

```diff
@@ -0,0 +1,543 @@
+			nss.Insert(ns)
+		}
+	}
+	nsList := a.desiredNamespaces.UnsortedList()
+	sort.Strings(nsList)
+	annotations[ApplySetAdditionalNamespacesAnnotation] = strings.Join(nsList, ",")
+	return annotations, nss, gks
+}
+
+func (a *applySet) apply(ctx context.Context, dryRun bool) (*ApplyResult, error) {
+	results := &ApplyResult{DesiredCount: a.desired.Len()}
+
+	// If dryRun is true, we will not update the parent labels and annotations.
+	if !dryRun {
+		parent, err := meta.Accessor(a.parent)
+		if err != nil {
+			return results, fmt.Errorf("unable to get parent: %w", err)
+		}
+		// Record the current labels and annotations
+		a.currentLabels = parent.GetLabels()
+		a.currentAnnotations = parent.GetAnnotations()
+
+		// We will ensure the parent is updated with the latest applyset before applying the resources.
+		a.supersetNamespaces, a.supersetGKs, err = a.updateParentLabelsAndAnnotations(ctx, updateToSuperset)
+		if err != nil {
+			return results, fmt.Errorf("unable to update Parent: %w", err)
+		}
+	}
+
+	options := a.applyOptions
+	if dryRun {
+		options.DryRun = []string{"All"}
+	}
+	for _, obj := range a.desired.objects {
+
+		dynResource := a.resourceClient(obj)
+		// This should never happen, but if it does, we want to know about it.
+		if dynResource == nil {
+			return results, fmt.Errorf("FATAL: rest mapping not found for %v", obj.GroupVersionKind())
+		}
+		lastApplied, err := dynResource.Apply(ctx, obj.GetName(), obj.Unstructured, options)
+		ao := results.recordApplied(obj, lastApplied, err)
+		a.log.V(2).Info("applied object", "object", obj.String(), "applied-revision", lastApplied.GetResourceVersion(),
+			"error", err)
+
+		if a.afterApplyFn != nil {
+			if err := a.afterApplyFn(ao); err != nil {
+				return results, fmt.Errorf("error after apply: %w", err)
+			}
+		}
+
+		// TODO: Implement health computation after apply
+		//message := ""
+		//tracker.isHealthy, message, err = a.computeHealth(lastApplied)
+		//results.reportHealth(gvk, nn, lastApplied, tracker.isHealthy, message, err)
+	}
+
+	return results, nil
+}
+
+func (a *applySet) prune(ctx context.Context, results *ApplyResult, dryRun bool) (*ApplyResult, error) {
+	pruneObjects, err := a.FindAllObjectsToPrune(ctx, a.dynamicClient, results.AppliedUIDs())
+	if err != nil {
+		return results, err
+	}
+	options := a.deleteOptions
+	if dryRun {
+		options.DryRun = []string{"All"}
+	}
+	for i := range pruneObjects {
+		name := pruneObjects[i].GetName()
+		namespace := pruneObjects[i].GetNamespace()
+		mapping := pruneObjects[i].Mapping
+
+		// TODO: Why is it always using namesapce ?
+		if a.beforePruneFn != nil {
+			if err := a.beforePruneFn(PrunedObject{
+				PruneObject: pruneObjects[i],
+			}); err != nil {
+				return results, fmt.Errorf("error from before-prune callback: %w", err)
+			}
+		}
+		err := a.dynamicClient.Resource(mapping.Resource).Namespace(namespace).Delete(ctx, name, options)
```

Comment:
does this need something like

```
if mapping.Scope.Name() == meta.RESTScopeNameNamespace {
		err := a.dynamicClient.Resource(mapping.Resource).Namespace(namespace).Delete(ctx, name, options)
	} else {
		err := a.dynamicClient.Resource(mapping.Resource).Delete(ctx, name, options)
}
```
if this doesnt then the code in `New` above also doesnt need it I think

Reply:
added the recommended code to check namespacescope